### PR TITLE
Added MPNowPlayingInfo for player

### PR DESCRIPTION
**Pull request by doctor3w**
Adds lock screen now playing info and commands. There are bugs that need to be solved once we make player live forever.

### CODE REVIEW

File: Recast/AppDelegate.swift [hunk 1]
```diff
@@ -80,6 +78,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     @objc func beginInterruption() {
         // TODO: handle audio interruptions
+        // need global player
```
Comment (mindylou) [inline Recast/AppDelegate.swift:81]: what do you mean by global player?

File: Recast/AppDelegate.swift [hunk 2]
```diff
@@ -80,6 +78,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+    }
+
+    // handles headphone events
+    override func remoteControlReceived(with event: UIEvent?) {
+        super.remoteControlReceived(with: event)
+        if let e = event, e.type == .remoteControl {
+            switch e.subtype {
+            case .remoteControlPlay:
+//                Player.sharedInstance.play()
```
Comment (mindylou) [inline Recast/AppDelegate.swift:90]: why are these commented out?

File: Recast/PlayerViewController.swift [hunk 3]
```diff
@@ -328,14 +330,79 @@ class PlayerViewController: UIViewController {
...
-        // TODO: Implement
+    // Configures the Remote Command Center for our player. Should only be called once (in init)
+    func configureCommands() {
+        UIApplication.shared.beginReceivingRemoteControlEvents()
+        let commandCenter = MPRemoteCommandCenter.shared()
+        commandCenter.pauseCommand.addTarget(self, action: #selector(playPauseButtonWasPressed(_:)))
+        commandCenter.playCommand.addTarget(self, action: #selector(playPauseButtonWasPressed(_:)))
+        commandCenter.skipForwardCommand.addTarget(self, action: #selector(skipForwardButtonWasPressed(_:)))
+        commandCenter.skipBackwardCommand.addTarget(self, action: #selector(skipBackButtonWasPressed(_:)))
+        commandCenter.skipForwardCommand.preferredIntervals = [30]
```
Comment (mindylou) [inline Recast/PlayerViewController.swift:340]: do we want to consider making 30 a constant?

File: Recast/PlayerViewController.swift [hunk 4]
```diff
@@ -328,14 +330,79 @@ class PlayerViewController: UIViewController {
...
     }
 
     func updateNowPlayingInfo() {
-        if let imageUrl = current?.podcast?.artworkUrl600, episodeImageView != nil {
-            episodeImageView.kf.setImage(with: imageUrl)
+        guard let episode = current, let podcast = episode.podcast else {
+            configureNowPlaying(info: nil)
+            return
+        }
+        print("Updated")
```
Comment (mindylou) [inline Recast/PlayerViewController.swift]: remove print

**Review by jackthmp — DISMISSED** [review]
LGTM after you make mindy's changes

File: Recast/AppDelegate.swift (hunk 1 above)
Comment (doctor3w) [reply Recast/AppDelegate.swift:81]: Need player to live forever, currently just lives when you open it

File: Recast/AppDelegate.swift (hunk 2 above)
Comment (doctor3w) [reply Recast/AppDelegate.swift:90]: Need global player, we don't have singleton anymore, so reference to the view controller

File: Recast/PlayerViewController.swift (hunk 3 above)
Comment (doctor3w) [reply Recast/PlayerViewController.swift:340]: I was hoping that we'd make it some sort of setting, so for now I think just 30 is fine.

**Pull request merged**